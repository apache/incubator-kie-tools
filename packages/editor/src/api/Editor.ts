/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Notification } from "@kie-tooling-core/notifications/dist/api";
import { Rect } from "@kie-tooling-core/guided-tour/dist/api";

/**
 * Editor component API. Implement this class to create an Editor.
 */
export interface Editor extends EditorApi {
  af_isReact: boolean;
  af_componentId: string;
  af_componentTitle: string;

  af_componentRoot(): React.ReactPortal | React.ReactElement | HTMLElement | string;
  af_onStartup?(): void;
  af_onOpen?(): void;
}

/**
 * Editor component API. Basic Editor feature definitions.
 */
export interface EditorApi {
  setContent(path: string, content: string): Promise<void>;
  getContent(): Promise<string>;
  getPreview(): Promise<string | undefined>;
  getElementPosition(selector: string): Promise<Rect | undefined>;
  undo(): Promise<void>;
  redo(): Promise<void>;
  validate(): Promise<Notification[]>;
}
