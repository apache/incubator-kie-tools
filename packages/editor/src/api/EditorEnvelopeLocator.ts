/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { IMinimatch, Minimatch } from "minimatch";

export enum EnvelopeContentType {
  PATH,
  CONTENT,
}

export type EnvelopeContent =
  | {
      type: EnvelopeContentType.PATH;
      path: string;
    }
  | {
      type: EnvelopeContentType.CONTENT;
      content: string;
    };

export class EnvelopeMapping {
  public matcher: IMinimatch;

  constructor(
    private readonly args: {
      type: string;
      filePathGlob: string;
      resourcesPathPrefix: string;
      envelopeContent: EnvelopeContent;
    }
  ) {
    this.matcher = new Minimatch(args.filePathGlob, { nocase: true, dot: true });
  }

  get type(): string {
    return this.args.type;
  }

  get filePathGlob(): string {
    return this.args.filePathGlob;
  }

  get resourcesPathPrefix(): string {
    return this.args.resourcesPathPrefix;
  }

  get envelopeContent(): EnvelopeContent {
    return this.args.envelopeContent;
  }
}

export class EditorEnvelopeLocator {
  constructor(public readonly targetOrigin: string, public readonly envelopeMappings: EnvelopeMapping[]) {}

  public getEnvelopeMapping(path: string) {
    return this.envelopeMappings.find((mapping) => {
      return mapping.matcher.match(path);
    });
  }

  public hasMappingFor(path: string) {
    return this.getEnvelopeMapping(path) !== undefined;
  }
}
