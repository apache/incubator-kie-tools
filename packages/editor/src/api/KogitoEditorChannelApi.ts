/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { CapabilityChannelApi } from "@kie-tooling-core/backend/dist/channel-api";
import { NotificationsApi } from "@kie-tooling-core/notifications/dist/api";
import { GuidedTourChannelApi } from "@kie-tooling-core/guided-tour/dist/api";
import { I18nChannelApi } from "@kie-tooling-core/i18n/dist/api";
import { WorkspaceApi } from "@kie-tooling-core/workspace/dist/api";
import { EditorContent } from "./EditorContent";
import { StateControlCommand } from "./StateControlCommand";

export interface KogitoEditorChannelApi
  extends GuidedTourChannelApi,
    I18nChannelApi,
    CapabilityChannelApi,
    WorkspaceApi,
    NotificationsApi {
  kogitoEditor_ready(): void;
  kogitoEditor_setContentError(content: EditorContent): void;
  kogitoEditor_stateControlCommandUpdate(command: StateControlCommand): void;
  kogitoEditor_contentRequest(): Promise<EditorContent>;
}
