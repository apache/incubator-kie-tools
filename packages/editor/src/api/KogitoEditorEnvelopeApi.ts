/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { EditorContent } from "./EditorContent";
import { KeyboardShortcutsEnvelopeApi } from "@kie-tooling-core/keyboard-shortcuts/dist/api";
import { GuidedTourEnvelopeApi } from "@kie-tooling-core/guided-tour/dist/api";
import { I18nEnvelopeApi } from "@kie-tooling-core/i18n/dist/api";
import { Notification } from "@kie-tooling-core/notifications/dist/api";

export interface Association {
  origin: string;
  envelopeServerId: string;
}

export enum ChannelType {
  VSCODE = "VSCODE",
  ONLINE = "ONLINE",
  GITHUB = "GITHUB",
  DESKTOP = "DESKTOP",
  EMBEDDED = "EMBEDDED",
  OTHER = "OTHER",
}

export interface EditorInitArgs {
  resourcesPathPrefix: string;
  fileExtension: string;
  initialLocale: string;
  isReadOnly: boolean;
  channel: ChannelType;
}

export interface KogitoEditorEnvelopeApi extends KeyboardShortcutsEnvelopeApi, GuidedTourEnvelopeApi, I18nEnvelopeApi {
  kogitoEditor_contentChanged(content: EditorContent): Promise<void>;
  kogitoEditor_editorUndo(): void;
  kogitoEditor_editorRedo(): void;
  kogitoEditor_initRequest(association: Association, editorInit: EditorInitArgs): Promise<void>;
  kogitoEditor_contentRequest(): Promise<EditorContent>;
  kogitoEditor_previewRequest(): Promise<string>;
  kogitoEditor_validate(): Promise<Notification[]>;
}
