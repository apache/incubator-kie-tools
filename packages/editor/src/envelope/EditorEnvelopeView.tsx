/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { Editor } from "../api";
import { LoadingScreen } from "./LoadingScreen";
import { KeyBindingsHelpOverlay } from "./KeyBindingsHelpOverlay";
import { useCallback, useImperativeHandle, useState } from "react";

interface Props {
  setLocale: React.Dispatch<string>;
}

export interface EditorEnvelopeViewApi<E extends Editor> {
  getEditor: () => E | undefined;
  setEditor: (editor: E) => void;
  setLoading: () => void;
  setLoadingFinished: () => void;
  setLocale: (locale: string) => void;
}

export const EditorEnvelopeViewRef: React.ForwardRefRenderFunction<EditorEnvelopeViewApi<Editor>, Props> = (
  props: Props,
  forwardingRef
) => {
  const [editor, setEditor] = useState<Editor | undefined>(undefined);
  const [loading, setLoading] = useState(true);

  const getEditor = useCallback(() => {
    return editor;
  }, [editor]);

  const setNewEditor = useCallback((newEditor: Editor) => {
    setEditor(newEditor);
  }, []);

  const setLoadingInit = useCallback(() => {
    setLoading(true);
  }, []);

  const setLoadingFinished = useCallback(() => {
    setLoading(false);
  }, []);

  const setLocale = useCallback((locale: string) => {
    props.setLocale(locale);
  }, []);

  useImperativeHandle(
    forwardingRef,
    () => {
      return {
        getEditor: () => getEditor(),
        setEditor: (newEditor) => setNewEditor(newEditor),
        setLoading: () => setLoadingInit(),
        setLoadingFinished: () => setLoadingFinished(),
        setLocale: (locale) => setLocale(locale),
      };
    },
    []
  );

  return (
    <>
      {!loading && <KeyBindingsHelpOverlay />}
      <LoadingScreen loading={loading} />
      <div style={{ position: "absolute", width: "100vw", height: "100vh", top: "0", left: "0" }}>
        {editor && editor.af_isReact && editor.af_componentRoot()}
      </div>
    </>
  );
};

export const EditorEnvelopeView = React.forwardRef(EditorEnvelopeViewRef);
