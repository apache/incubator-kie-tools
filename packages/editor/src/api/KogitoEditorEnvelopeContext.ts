/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { useContext } from "react";
import { ApiDefinition, MessageBusClientApi } from "@kie-tooling-core/envelope-bus/dist/api";
import { KogitoEditorChannelApi } from "./KogitoEditorChannelApi";
import { DefaultKeyboardShortcutsService } from "@kie-tooling-core/keyboard-shortcuts/dist/envelope";
import { I18nService } from "@kie-tooling-core/i18n/dist/envelope";
import { OperatingSystem } from "@kie-tooling-core/operating-system";

export interface KogitoEditorEnvelopeContextType<
  ChannelApi extends KogitoEditorChannelApi & ApiDefinition<ChannelApi>
> {
  channelApi: MessageBusClientApi<ChannelApi>;
  operatingSystem?: OperatingSystem;
  services: {
    keyboardShortcuts: DefaultKeyboardShortcutsService;
    guidedTour: { isEnabled: () => boolean };
    i18n: I18nService;
  };
}

export const KogitoEditorEnvelopeContext = React.createContext<KogitoEditorEnvelopeContextType<any>>({} as any);

export function useKogitoEditorEnvelopeContext<
  ChannelApi extends KogitoEditorChannelApi & ApiDefinition<ChannelApi> = KogitoEditorChannelApi
>() {
  return useContext(KogitoEditorEnvelopeContext) as KogitoEditorEnvelopeContextType<ChannelApi>;
}
