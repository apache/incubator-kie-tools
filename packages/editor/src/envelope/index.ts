/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import "@patternfly/react-core/dist/styles/base.css";
import "./styles.scss";
import {
  Editor,
  EditorFactory,
  KogitoEditorChannelApi,
  KogitoEditorEnvelopeApi,
  KogitoEditorEnvelopeContextType,
} from "../api";
import { KogitoEditorEnvelope } from "./KogitoEditorEnvelope";
import { ApiDefinition, EnvelopeBus } from "@kie-tooling-core/envelope-bus/dist/api";
import { KogitoEditorEnvelopeApiImpl } from "./KogitoEditorEnvelopeApiImpl";
import { DefaultKeyboardShortcutsService } from "@kie-tooling-core/keyboard-shortcuts/dist/envelope";
import { I18nService } from "@kie-tooling-core/i18n/dist/envelope";
import { Envelope, EnvelopeApiFactory } from "@kie-tooling-core/envelope";
import { EditorEnvelopeViewApi } from "./EditorEnvelopeView";
import { getOperatingSystem } from "@kie-tooling-core/operating-system";

/**
 * Starts the Editor envelope at a given container. Uses `bus` to send messages out of the Envelope and creates Editors based on the editorFactory provided.
 * @param args.container The DOM element where the envelope should be rendered.
 * @param args.bus The implementation of EnvelopeBus to send messages out of the envelope.
 * @param args.editorFactory The factory of Editors provided by this EditorEnvelope.
 */
export function init(args: {
  container: HTMLElement;
  bus: EnvelopeBus;
  editorFactory: EditorFactory<Editor, KogitoEditorChannelApi>;
}) {
  initCustom({
    container: args.container,
    bus: args.bus,
    apiImplFactory: {
      create: (createArgs) => new KogitoEditorEnvelopeApiImpl(createArgs, args.editorFactory),
    },
  });
}

export function initCustom<
  E extends Editor,
  EnvelopeApi extends KogitoEditorEnvelopeApi & ApiDefinition<EnvelopeApi>,
  ChannelApi extends KogitoEditorChannelApi & ApiDefinition<ChannelApi>
>(args: {
  container: HTMLElement;
  bus: EnvelopeBus;
  apiImplFactory: EnvelopeApiFactory<
    EnvelopeApi,
    ChannelApi,
    EditorEnvelopeViewApi<E>,
    KogitoEditorEnvelopeContextType<ChannelApi>
  >;
}) {
  const defaultKeyboardShortcuts = new DefaultKeyboardShortcutsService({ os: getOperatingSystem() });
  const i18nService = new I18nService();
  const envelope = new Envelope<
    EnvelopeApi,
    ChannelApi,
    EditorEnvelopeViewApi<E>,
    KogitoEditorEnvelopeContextType<ChannelApi>
  >(args.bus);

  return new KogitoEditorEnvelope(args.apiImplFactory, defaultKeyboardShortcuts, i18nService, envelope).start(
    args.container
  );
}
export { EditorEnvelopeViewApi } from "./EditorEnvelopeView";
export * from "./KogitoEditorEnvelopeApiImpl";
