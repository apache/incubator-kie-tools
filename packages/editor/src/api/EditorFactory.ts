/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { EditorInitArgs } from "./KogitoEditorEnvelopeApi";
import { Editor } from "./Editor";
import { KogitoEditorEnvelopeContextType } from "./KogitoEditorEnvelopeContext";
import { ApiDefinition } from "@kie-tooling-core/envelope-bus/dist/api";
import { KogitoEditorChannelApi } from "./KogitoEditorChannelApi";

/**
 * Factory of Editors to be created inside the envelope.
 */
export interface EditorFactory<
  E extends Editor,
  ChannelApi extends KogitoEditorChannelApi & ApiDefinition<ChannelApi>
> {
  /**
   * Returns an Editor instance.
   * @param envelopeContext The context to be used by Editor implementation.
   * @param initArgs Initial arguments required for the Editor to initialize itself properly.
   */
  createEditor(envelopeContext: KogitoEditorEnvelopeContextType<ChannelApi>, initArgs: EditorInitArgs): Promise<E>;
}
