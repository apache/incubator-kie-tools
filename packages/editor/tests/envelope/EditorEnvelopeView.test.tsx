/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { cleanup, fireEvent, getByTestId, render, act } from "@testing-library/react";
import { EditorEnvelopeView, EditorEnvelopeViewApi } from "@kie-tooling-core/editor/dist/envelope/EditorEnvelopeView";
import { DummyEditor } from "./DummyEditor";
import { usingEditorEnvelopeI18nContext, usingEnvelopeContext } from "./utils";
import { Editor } from "@kie-tooling-core/editor/dist/api";

function renderEditorEnvelopeView(): EditorEnvelopeViewApi<Editor> {
  const editorEnvelopeRef = React.createRef<EditorEnvelopeViewApi<Editor>>();
  const setLocale = jest.fn();
  render(
    usingEditorEnvelopeI18nContext(
      usingEnvelopeContext(<EditorEnvelopeView ref={editorEnvelopeRef} setLocale={setLocale} />).wrapper
    ).wrapper
  );
  return editorEnvelopeRef.current!;
}

describe("EditorEnvelopeView", () => {
  afterEach(() => {
    cleanup();
  });

  test("first open", async () => {
    const _ = renderEditorEnvelopeView();
    expect(document.body).toMatchSnapshot();
  });

  test("after loading stops", async () => {
    const view = renderEditorEnvelopeView();

    act(() => {
      view.setLoadingFinished();
    });

    fireEvent.animationEnd(getByTestId(document.body, "loading-screen-div"));

    expect(document.body).toMatchSnapshot();
  });

  test("after loading stops and editor is set", () => {
    const view = renderEditorEnvelopeView();

    act(() => {
      view.setLoadingFinished();
    });

    fireEvent.animationEnd(getByTestId(document.body, "loading-screen-div"));

    act(() => {
      view.setEditor(new DummyEditor());
    });

    expect(document.body).toMatchSnapshot();
  });

  test("after set content", () => {
    const view = renderEditorEnvelopeView();

    act(() => {
      view.setLoadingFinished();
    });

    fireEvent.animationEnd(getByTestId(document.body, "loading-screen-div"));

    act(() => {
      view.setEditor(new DummyEditor());
    });
    setTimeout(() => view.getEditor()!.setContent("/some/path.txt", "some-test-content"), 100);

    expect(document.body).toMatchSnapshot();
  });
});
