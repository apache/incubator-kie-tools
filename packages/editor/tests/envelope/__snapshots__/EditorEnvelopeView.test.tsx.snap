// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorEnvelopeView after loading stops 1`] = `
<body>
  <div>
    <div
      class="kogito-tooling--keyboard-shortcuts kogito-tooling--keyboard-shortcuts-icon"
      data-testid="keyboard-shortcuts-help-overlay-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 576 512"
        width="1em"
      >
        <path
          d="M528 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM128 180v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm288 0v-40c0-6.627-5.373-12-12-12H172c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h232c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12z"
        />
      </svg>
    </div>
    <div
      style="position: absolute; width: 100vw; height: 100vh; top: 0px; left: 0px;"
    />
  </div>
  <div />
</body>
`;

exports[`EditorEnvelopeView after loading stops and editor is set 1`] = `
<body>
  <div>
    <div
      class="kogito-tooling--keyboard-shortcuts kogito-tooling--keyboard-shortcuts-icon"
      data-testid="keyboard-shortcuts-help-overlay-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 576 512"
        width="1em"
      >
        <path
          d="M528 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM128 180v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm288 0v-40c0-6.627-5.373-12-12-12H172c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h232c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12z"
        />
      </svg>
    </div>
    <div
      style="position: absolute; width: 100vw; height: 100vh; top: 0px; left: 0px;"
    >
      <div
        data-testid="dummy-editor"
      >
        Here's the dummy content: 
        
      </div>
    </div>
  </div>
  <div />
</body>
`;

exports[`EditorEnvelopeView after set content 1`] = `
<body>
  <div>
    <div
      class="kogito-tooling--keyboard-shortcuts kogito-tooling--keyboard-shortcuts-icon"
      data-testid="keyboard-shortcuts-help-overlay-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 576 512"
        width="1em"
      >
        <path
          d="M528 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h480c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM128 180v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm-336 96v-40c0-6.627-5.373-12-12-12H76c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12zm288 0v-40c0-6.627-5.373-12-12-12H172c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h232c6.627 0 12-5.373 12-12zm96 0v-40c0-6.627-5.373-12-12-12h-40c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h40c6.627 0 12-5.373 12-12z"
        />
      </svg>
    </div>
    <div
      style="position: absolute; width: 100vw; height: 100vh; top: 0px; left: 0px;"
    >
      <div
        data-testid="dummy-editor"
      >
        Here's the dummy content: 
        
      </div>
    </div>
  </div>
  <div />
</body>
`;

exports[`EditorEnvelopeView first open 1`] = `
<body>
  <div>
    <div
      class="kogito-tooling--loading-screen"
      id="loading-screen"
    >
      <div
        class="kogito-tooling--loading-screen "
        data-testid="loading-screen-div"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            class="kogito-tooling--loading-screen-spinner"
          >
            <div>
              <span
                aria-valuetext="Loading..."
                class="pf-c-spinner pf-m-xl"
                role="progressbar"
              >
                <span
                  class="pf-c-spinner__clipper"
                />
                <span
                  class="pf-c-spinner__lead-ball"
                />
                <span
                  class="pf-c-spinner__tail-ball"
                />
              </span>
            </div>
            <h5
              class="pf-c-title pf-m-md"
            >
              Loading...
            </h5>
          </div>
        </div>
      </div>
    </div>
    <div
      style="position: absolute; width: 100vw; height: 100vh; top: 0px; left: 0px;"
    />
  </div>
</body>
`;
